src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -113,10 +113,22 @@ public SequencePosition Position
         {
             get
             {
-                // TODO: Cannot use Slice even though it would be faster: https://github.com/dotnet/corefx/issues/33291
-                return _isInputSequence
-                    ? _sequence.GetPosition(BytesConsumed)
-                    : default;
+                if (_isInputSequence)
+                {
+                    if (_isSingleSegment)
```

Comment:
Does this extra branch help things? Do you not have a _currentPosition in this case?

Reply:
No, we don't. `_currentPosition` is set to default for `_isSingleSegment` case since we go down the span code path.

https://github.com/dotnet/corefx/blob/91277c6149f56f7f0701303cfa616d95b418ce3a/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L54-L61

I could set `_currentPosition` to `jsonData.Start` but I didn't see a need for it.

Reply:
Yea, do that. Avoid those extra needless branches.